File: packages/java/engine/src/test/snapshot/examples/linear_search.java
package com.algoflow.runner;

public class Main {
    private int[] arr = {5, 3, 8, 1, 9, 2, 7};

    public static void main(String[] args) {
        new Main().linearSearch(9);
    }

    public int linearSearch(int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) return i;
        }
        return -1;
    }
}

File: packages/python/engine/test/snapshot/examples/linear_search.py
arr = [5, 3, 8, 1, 9, 2, 7]

def linear_search(a, target):
    for i in range(len(a)):
        if a[i] == target:
            return i
    return -1

result = linear_search(arr, 9)
print("found at", result)
